jerry-core/parser/js/js-parser-expr.c

```diff
@@ -687,6 +687,11 @@ parser_parse_class_body (parser_context_t *context_p, /**< context */
         context_p->token.flags |= LEXER_NO_SKIP_SPACES;
         ident_opts |= LEXER_OBJ_IDENT_CLASS_PRIVATE;
         is_private = true;
+
+        if (lexer_check_next_character (context_p, LIT_CHAR_ASTERISK))
+        {
+          parser_raise_error (context_p, PARSER_ERR_INVALID_CHARACTER);
+        }
```

Comment:
There is a `if (is_private && context_p->token.type == LEXER_LITERAL)` below. Would it be a good idea to move it here?

Reply:
Moving this condition ruins the some jerry and 262 tests. Mainly those with private field names.